spinner/spinner_test.go

```diff
@@ -105,10 +116,63 @@ func TestSpinnerUpdate(t *testing.T) {
 	}
 }
 
-func TestAccessibleSpinner(t *testing.T) {
-	s := New().Accessible(true)
-	err := s.Run()
+func TestSpinnerSimple(t *testing.T) {
+	exercise(t, func() *Spinner {
+		return New().Action(func() {})
+	}, requireNoError)
+}
+
+func TestSpinnerWithContextAndAction(t *testing.T) {
+	exercise(t, func() *Spinner {
+		ctx := context.Background()
+		return New().Context(ctx).Action(func() {})
+	}, requireNoError)
+}
+
+func TestSpinnerWithActionError(t *testing.T) {
+	fake := errors.New("fake")
+	exercise(t, func() *Spinner {
+		return New().ActionWithErr(func(context.Context) error { return fake })
+	}, requireErrorIs(fake))
+}
+
+func exercise(t *testing.T, factory func() *Spinner, checker func(tb testing.TB, err error)) {
+	t.Helper()
+	t.Run("accessible", func(t *testing.T) {
+		err := factory().
+			Accessible(true).
+			Output(io.Discard).
+			Run()
+		checker(t, err)
+	})
+	t.Run("regular", func(t *testing.T) {
+		err := factory().
+			Accessible(false).
+			Output(io.Discard).
+			Run()
+		checker(t, err)
+	})
+}
+
+func requireNoError(tb testing.TB, err error) {
+	tb.Helper()
 	if err != nil {
-		t.Errorf("Run() in accessible mode returned an error: %v", err)
+		tb.Errorf("expected no error, got %v", err)
+	}
+}
+
+func requireErrorIs(target error) func(tb testing.TB, err error) {
+	return func(tb testing.TB, err error) {
+		tb.Helper()
+		if !errors.Is(err, target) {
+			tb.Errorf("expected error to be %v, got %v", target, err)
+		}
+	}
+}
+
+func requireContextCanceled(tb testing.TB, err error) {
+	tb.Helper()
+	if !errors.Is(err, context.Canceled) && !errors.Is(err, tea.ErrProgramKilled) {
+		tb.Errorf("expected to get a context canceled error, got %v", err)
 	}
```

Comment:
A switch could be more readable


```suggestion
	switch {
		case errors.Is(err, context.Canceled):
		case errors.Is(err, tea.ErrProgramKilled):

		default:
			tb.Errorf("expected to get a context canceled error, got %v", err)
 	}
```

Reply:
I'm surprised you said 👍 but didn't apply while you did with my other 2 feedbacks.

But this one is not yet resolved, so I might simply wait 😅